plugin/pkg/scheduler/factory/factory.go

```diff
@@ -227,16 +237,49 @@ func (f *ConfigFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 		Algorithm:  algo,
 		Binder:     &binder{f.Client},
 		NextPod: func() *api.Pod {
-			pod := f.PodQueue.Pop().(*api.Pod)
-			glog.V(2).Infof("About to try and schedule pod %v", pod.Name)
-			return pod
+			return f.getNextPod()
 		},
+		//		NextPod:    f.getNextPod(),
 		Error:               f.makeDefaultErrorFunc(&podBackoff, f.PodQueue),
 		BindPodsRateLimiter: f.BindPodsRateLimiter,
 		StopEverything:      f.StopEverything,
 	}, nil
 }
 
+func (f *ConfigFactory) getNextPod() *api.Pod {
+	for {
+		pod := f.PodQueue.Pop().(*api.Pod)
+		if f.responsibleForPod(pod) {
+			glog.V(2).Infof("About to try and schedule pod %v", pod.Name)
+			return pod
+		}
+	}
+}
+
+func (f *ConfigFactory) responsibleForPod(pod *api.Pod) bool {
+	if f.SchedulerName == "default-scheduler" {
+		if pod.Annotations[SchedulerAnnotationKey] == f.SchedulerName {
+			return true
+		}
+		if pod.Annotations[SchedulerAnnotationKey] == "" {
+			return true
+		}
+		return false
+	} else {
+		if pod.Annotations[SchedulerAnnotationKey] == f.SchedulerName {
+			return true
+		}
+		return false
+	}
+	//	if pod.Annotations[SchedulerAnnotationKey] == f.SchedulerName {
```

Comment:
Sure, this is not the final version yet. I am working on a integration test. Maybe I can update the PR before the integration test is finished.